pkgs/by-name/mi/mimalloc/package.nix

```
@@ -34,15 +35,32 @@ stdenv.mkDerivation rec {
     cmake
     ninja
   ];
-  cmakeFlags =
-    [ "-DMI_INSTALL_TOPLEVEL=ON" ]
-    ++ lib.optionals secureBuild [ "-DMI_SECURE=ON" ]
-    ++ lib.optionals stdenv.hostPlatform.isStatic [ "-DMI_BUILD_SHARED=OFF" ]
-    ++ lib.optionals (!doCheck) [ "-DMI_BUILD_TESTS=OFF" ];
+  cmakeFlags = [
+    (lib.cmakeBool "MI_INSTALL_TOPLEVEL" true)
+    (lib.cmakeBool "DMI_SECURE" secureBuild)
+    (lib.cmakeBool "MI_BUILD_SHARED" (
+      !stdenv.hostPlatform.isStatic && stdenv.hostPlatform.hasSharedLibraries
+    ))
+    (lib.cmakeBool "MI_LIBC_MUSL" (stdenv.hostPlatform.libc == "musl"))
+    (lib.cmakeBool "MI_LOCAL_DYNAMIC_TLS" localDynamicTLS)
+    (lib.cmakeBool "MI_BUILD_TESTS" finalAttrs.doCheck)
+
+    # MI_OPT_ARCH is inaccurate (e.g. it assumes aarch64 == armv8.1-a).
+    # Nixpkgs's native platform configuration does a better job.
+    (lib.cmakeBool "MI_OPT_ARCH" false)
+  ];
```

Just an idea that I think might improve readability:

```suggestion
  cmakeFlags = lib.mapAttrsToList lib.cmakeBool {
    MI_INSTALL_TOPLEVEL = true;
    MI_SECURE = secureBuild;
    MI_BUILD_SHARED = !stdenv.hostPlatform.isStatic && stdenv.hostPlatform.hasSharedLibraries;
    MI_LIBC_MUSL = stdenv.hostPlatform.libc == "musl";
    MI_LOCAL_DYNAMIC_TLS = localDynamicTLS;
    MI_BUILD_TESTS = finalAttrs.doCheck;

    # MI_OPT_ARCH is inaccurate (e.g. it assumes aarch64 == armv8.1-a).
    # Nixpkgs's native platform configuration does a better job.
    MI_OPT_ARCH = false;
  };
```